<launch>
	<arg name="pkg_name" value="joy_mapper"/>
	<arg name="node_name" default="joy_mapper_node"/>
		
	<arg name="veh" doc="Name of vehicle. ex: megaman"/>
	<arg name="local" default="false" doc="true for running on laptop. false for running on vehicle."/>
	<arg name="config" default="baseline" doc="Specify a config."/>
	<arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman"/>
        <arg name="model"/>

	<!-- Run on local (laptop) -->
	<node if="$(arg local)" ns="$(arg veh)" pkg="ncs_following" type="ncs_joy_mapper_node.py" name="$(arg node_name)" output="screen" clear_params="true" required="true">
		<rosparam command="load" file="$(find duckietown)/config/$(arg config)/$(arg pkg_name)/$(arg node_name)/$(arg param_file_name).yaml"/>
        	<param name="model_name" type="str" value="$(arg model)"/>
	</node>

	<!-- Run on remote (vehicle) -->
	<include unless="$(arg local)" file="$(find duckietown)/machines"/>
	<node unless="$(arg local)" ns="$(arg veh)" machine="$(arg veh)" pkg="ncs_following" type="ncs_joy_mapper_node.py" name="$(arg node_name)" output="screen" clear_params="true" required="true">
		<rosparam command="load" file="$(find duckietown)/config/$(arg config)/$(arg pkg_name)/$(arg node_name)/$(arg param_file_name).yaml"/>
        	<param name="model_name" type="str" value="$(arg model)"/>
                <param name="ncs_pkg" type="str" value="$(find ncs_following)"/>
	</node>	

	<!-- Subscribe Topics -->
	<!-- "joy": sensor_msgs/Joy. The command read from joystick -->
	<!-- Publish Topics -->
	<!-- "~wheels_cmd": duckietown_msgs/WheelsCmd. Wheels command for duckiebot -->
</launch>
